plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -481,68 +481,80 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
 
     public static LongWriteFunction trinoDateToOracleDateWriteFunction()
     {
-        return new LongWriteFunction()
-        {
+        return new LongWriteFunction() {
             @Override
-            public String getBindExpression()
+            public void set(PreparedStatement statement, int index, long value) throws SQLException
             {
-                return "TO_DATE(?, 'SYYYY-MM-DD')";
```

Comment:
Please take a look at the diff around here (https://github.com/trinodb/trino/pull/12400/files) and try to make sure it contains the changes you wanted.

Reply:
cleaned up